glotaran/project/result.py

```diff
@@ -0,0 +1,181 @@
+    number_of_function_evaluations: int
+    """The number of function evaluations."""
+    initial_parameters: ParameterGroup
+    optimized_parameters: ParameterGroup
+    """The optimized parameters, organized in a :class:`ParameterGroup`"""
+    scheme: Scheme
+    success: bool
+    """Indicates if the optimization was successful."""
+    termination_reason: str
+    """The reason (message when) the optimizer terminated"""
+
+    # The below can be none in case of unsuccessful optimization
+    chi_square: float | None = None
+    r"""The chi-square of the optimization.
+
+    :math:`\chi^2 = \sum_i^N [{Residual}_i]^2`."""
+    covariance_matrix: ArrayLike | None = None
+    """Covariance matrix.
+
+    The rows and columns are corresponding to :attr:`free_parameter_labels`."""
+    degrees_of_freedom: int | None = None
+    """Degrees of freedom in optimization :math:`N - N_{vars}`."""
+    jacobian: ArrayLike | None = None
+    """Modified Jacobian matrix at the solution
+
+    See also: :func:`scipy.optimize.least_squares`
+    """
+    number_of_data_points: int | None = None
+    """Number of data points :math:`N`."""
+    number_of_jacobian_evaluations: int | None = None
+    """The number of jacobian evaluations."""
+    number_of_variables: int | None = None
+    """Number of variables in optimization :math:`N_{vars}`"""
+    optimality: float | None = None
+    reduced_chi_square: float | None = None
+    r"""The reduced chi-square of the optimization.
+
+    :math:`\chi^2_{red}= {\chi^2} / {(N - N_{vars})}`.
+    """
+    root_mean_square_error: float | None = None
+    r"""
+    The root mean square error the optimization.
+
+    :math:`rms = \sqrt{\chi^2_{red}}`
+    """
+
+    @property
+    def model(self) -> Model:
+        return self.scheme.model
+
+    def get_scheme(self) -> Scheme:
+        """Return a new scheme from the Result object with optimized parameters.
+
+        Returns
+        -------
+        Scheme
+            A new scheme with the parameters set to the optimized values.
+            For the dataset weights the (precomputed) weights from the original scheme are used.
+        """
+        data = {}
+
+        for label, dataset in self.data.items():
+            data[label] = dataset.data.to_dataset(name="data")
+            if "weight" in dataset:
+                data[label]["weight"] = dataset.weight
+
+        return Scheme(
+            model=self.model,
+            parameters=self.optimized_parameters,
+            data=data,
+            group_tolerance=self.scheme.group_tolerance,
+            non_negative_least_squares=self.scheme.non_negative_least_squares,
+            maximum_number_function_evaluations=self.scheme.maximum_number_function_evaluations,
+            ftol=self.scheme.ftol,
+            gtol=self.scheme.gtol,
+            xtol=self.scheme.xtol,
+            optimization_method=self.scheme.optimization_method,
+        )
+
+    def markdown(self, with_model=True) -> str:
```

Comment:
We could just use [tabulate](https://pypi.org/project/tabulate/) for table formatting and the code would be much easier to read.

Reply:
yes